infrastructure/json/src/main/java/tech/pegasys/teku/infrastructure/json/types/DeserializableArrayTypeDefinition.java

```diff
@@ -54,6 +54,9 @@ public CollectionT deserialize(final JsonParser parser) throws IOException {
     final List<ItemT> result = new ArrayList<>();
     while (parser.nextToken() != JsonToken.END_ARRAY) {
       result.add(itemType.deserialize(parser));
+      if (parser.getCurrentToken().isStructStart()) {
+        parser.skipChildren();
+      }
```

Comment:
hmm, this doesn't seem right - I'd have expected that if we get a nested array/object that we didn't expect we should fail, not just skip it.

All the tests seem to pass even without it so I'm not sure what case it's trying to handle.